src/ui/popupmenu/completion_item_widget.rs

```diff
@@ -37,11 +40,21 @@ impl CompletionItemWidgetWrap {
         let grid = gtk::Grid::new();
         grid.set_column_spacing(10);
 
-        let buf = get_icon_pixbuf(&item.kind.as_str(), icon_fg, size);
-        let kind = gtk::Image::new_from_pixbuf(&buf);
-        kind.set_tooltip_text(format!("kind: '{}'", item.kind).as_str());
-        kind.set_margin_start(margin);
-        grid.attach(&kind, 0, 0, 1, 1);
+        let buf = get_icon_pixbuf(&item.kind, icon_fg, size);
+        let image = match &item.kind {
+            CompletionItemKind::Unknown => {
+                // If another item in the popupmenu has a known kind,
+                // give kind_image an icon.
+                if items.iter().any(|item| !item.kind.is_unknown()) {
+                    gtk::Image::new_from_pixbuf(&buf) 
+                } else { gtk::Image::new() }
+            }
+            _ => gtk::Image::new_from_pixbuf(&buf),
+        };
+
+        // kind_image.set_tooltip_text(format!("kind: '{}'", &item.kind.get_kind()).as_str());
```

Comment:
@vhakulinen What should I do here about setting the tooltip text for the image?

Reply:
I think you can just add `kind_raw` to the `CompletionItem` struct and then use that to add the tooltip text. The tooltip text is here for debugging/troubleshooting.